executor/join_test.go

```diff
@@ -54,6 +54,15 @@ func (s *testSuiteJoin1) TestJoinPanic(c *C) {
 	tk.MustQuery("SELECT * FROM events e JOIN (SELECT MAX(clock) AS clock FROM events e2 GROUP BY e2.source) e3 ON e3.clock=e.clock")
 	err := tk.ExecToErr("SELECT * FROM events e JOIN (SELECT clock FROM events e2 GROUP BY e2.source) e3 ON e3.clock=e.clock")
 	c.Check(err, NotNil)
+
+	// Test for PR 18983, use to detect race.
+	tk.MustExec("use test")
```

Comment:
I have tried running this on different TiDB versions, and it is consistently green after and *before* the fix. I understand that this is a race condition bug and it only happens occasionally, but I doubt that the test that only fails once in a lifetime is really that useful.

Reply:
Sorry for that,  this test is really hard to reproduce, I haven't found a stable reproduction method yet...